### PR TITLE
Assert compressed keys are strictly shorter than regular

### PR DESCRIPTION
**Pull request by Empact**
This is a stricter and more meaningful test.

### CODE REVIEW

File: src/key.h [hunk 1]
```diff
@@ -37,8 +37,8 @@ class CKey
      * script supports up to 75 for single byte push
      */
     static_assert(
-        PRIVATE_KEY_SIZE >= COMPRESSED_PRIVATE_KEY_SIZE,
-        "COMPRESSED_PRIVATE_KEY_SIZE is larger than PRIVATE_KEY_SIZE");
+        PRIVATE_KEY_SIZE > COMPRESSED_PRIVATE_KEY_SIZE,
+        "COMPRESSED_PRIVATE_KEY_SIZE should be smaller than PRIVATE_KEY_SIZE");
```
Comment (fivepiece) [inline src/key.h]: > should be smaller
Compressed private keys are indeed larger than uncompressed ones.

File: src/pubkey.h [hunk 2]
```diff
@@ -42,9 +42,11 @@ class CPubKey
      * script supports up to 75 for single byte push
      */
     static_assert(
-        PUBLIC_KEY_SIZE >= COMPRESSED_PUBLIC_KEY_SIZE,
-        "COMPRESSED_PUBLIC_KEY_SIZE is larger than PUBLIC_KEY_SIZE");
-
+        PUBLIC_KEY_SIZE > COMPRESSED_PUBLIC_KEY_SIZE,
+        "COMPRESSED_PUBLIC_KEY_SIZE should be smaller than PUBLIC_KEY_SIZE");
+    static_assert(
+        SIGNATURE_SIZE > COMPACT_SIGNATURE_SIZE,
```
Comment (fivepiece) [inline src/pubkey.h]: Is this a DER encoded signature vs. a 64kb (r,s)?  In this case this is not always correct as DER encoded signatures can be very small.

File: src/key.h (hunk 1 above)
Comment (Empact) [reply src/key.h]: This test is relative to the *_SIZE constant values, which are defined above and do have this relationship.

Comment (fivepiece) [reply src/key.h]: Ah I see, should've expanded the lines :)

**Comment by practicalswift** [comment]
Concept ACK

**Comment by Empact** [comment]
Closing as insignificant.

**Pull request closed without merging**